src/main/java/de/cas/mse/exercise/csvdb/impl/AddressDb.java

```diff
@@ -9,80 +9,68 @@
 import java.util.Optional;
 import java.util.UUID;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import de.cas.mse.exercise.csvdb.CsvDB;
+import de.cas.mse.exercise.csvdb.PersistenceLayer;
+import de.cas.mse.exercise.csvdb.Record;
 import de.cas.mse.exercise.csvdb.data.Address;
 
-public class AddressDb implements CsvDB<Address> {
+public class AddressDb<T extends PersistenceLayer> implements CsvDB<Address> {
```

Comment:
Do you really need generics? T could be PersistenceLayer on its own maybe.

Reply:
Right. Not on my A-game on this exercise.